src/main/java/games/strategy/engine/data/properties/ComboProperty.java

```diff
@@ -36,15 +33,19 @@ public ComboProperty(final String name, final String description, final T defaul
   private ComboProperty(final String name, final String description, final T defaultValue,
       final Collection<T> possibleValues, final boolean allowNone) {
     super(name, description);
-    if (!allowNone && !possibleValues.contains(defaultValue) && defaultValue == null) {
-      throw new IllegalStateException("possible values does not contain default");
-    } else if (allowNone && !possibleValues.contains(defaultValue) && !possibleValues.isEmpty()) {
-      value = possibleValues.iterator().next();
-    } else if (allowNone && !possibleValues.contains(defaultValue)) {
-      try {
-        value = (T) "";
-      } catch (final Exception e) {
-        value = null;
+    if (!allowNone) {
+      if (defaultValue == null && !possibleValues.contains(defaultValue)) {
+        throw new IllegalStateException("possible values does not contain default");
+      }
+    } else if (!possibleValues.contains(defaultValue)) {
+      if (!possibleValues.isEmpty()) {
+        value = possibleValues.iterator().next();
+      } else {
+        try {
+          value = (T) "";
```

Comment:
Why is this even a thing?!?

Reply:
that is quite special : )
If `T` is a string then `value` will equal empty string, otherwise null. It's a really interesting way to express that : |